### PR TITLE
Bug 1214205 - editor must always use the original object namespace when trying to update

### PR DESCRIPTION
**Pull request by fabianofranz**

### CODE REVIEW

**Comment by fabianofranz** [comment]
@smarterclayton please review.

File: pkg/cmd/cli/cmd/edit.go [hunk 1]
```diff
@@ -211,7 +211,11 @@ func RunEdit(fullName string, f *clientcmd.Factory, out io.Writer, cmd *cobra.Co
 			if err != nil {
 				return err
 			}
-			updated, err := resource.NewHelper(info.Client, info.Mapping).Update(info.Namespace, info.Name, false, data)
+			namespace, err := info.ResourceMapping().Namespace(obj)
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/edit.go]: The info should already have the namespace on it.

Comment (fabianofranz) [inline pkg/cmd/cli/cmd/edit.go]: Isn't `info.Namespace` the *new* one, in case the user edited the namespace? That's the root of this issue.

Comment (smarterclayton) [inline pkg/cmd/cli/cmd/edit.go]: The original namespace is still available earlier in the code - comparing that gives you the effective error.

Comment (fabianofranz) [inline pkg/cmd/cli/cmd/edit.go]: If you mean `cmdNamespace` it's available here, but not in `applyPatch` and `AddError` (unless I pass it).

Comment (smarterclayton) [inline pkg/cmd/cli/cmd/edit.go]: Enforce it right after InfoForData - but remember, some objects don't have namespaces

Comment (fabianofranz) [inline pkg/cmd/cli/cmd/edit.go]: Please take a look.

File: pkg/cmd/cli/cmd/edit.go [hunk 2]
```diff
@@ -195,6 +195,13 @@ func RunEdit(fullName string, f *clientcmd.Factory, out io.Writer, cmd *cobra.Co
 			continue
 		}
 
+		err = resource.NewFlattenListVisitor(updates, rmap).Visit(func(info *resource.Info) error {
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/edit.go]: Use the existing DefaultNamespace visitor func

Comment (fabianofranz) [inline pkg/cmd/cli/cmd/edit.go]: @smarterclayton I believe you mean the `RequireNamespace` visitor func (sorry, this piece is new to me). Fixed, please review.

File: pkg/cmd/cli/cmd/edit.go [hunk 3]
```diff
@@ -195,6 +195,13 @@ func RunEdit(fullName string, f *clientcmd.Factory, out io.Writer, cmd *cobra.Co
 			continue
 		}
 
+		visitor := resource.NewFlattenListVisitor(updates, rmap)
+
+		// need to make sure the original namespace wasn't changed while editing
+		if err = visitor.Visit(resource.RequireNamespace(cmdNamespace)); err != nil {
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/edit.go]: If a user has made edits, they'll be lost here.  Need to do the same thing to preserve the user's edits.

Comment (fabianofranz) [inline pkg/cmd/cli/cmd/edit.go]: @smarterclayton Fixed.

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**